### PR TITLE
feat(dashboard): add breadcrumbs for ratelimits and authorization routes

### PR DESCRIPTION
**Pull request by guilhermerodz**
Adds breadcrumbs to the routes `ratelimits` and `authorization`

https://github.com/user-attachments/assets/d6082411-262c-4bcf-b5d0-5aa23a5b0ed0

### CODE REVIEW

File: apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+export const runtime = "edge";
+
+type PageProps = {
+  params: { permissionId: string };
+};
+
+async function AsyncPageBreadcrumb(props: PageProps) {
+  const tenantId = getTenantId();
+
+  const workspace = await db.query.workspaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx:25]: this query is super expensive and not cached, why do you need all of those relations?

Comment (chronark) [reply apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx:25]: all you need is the name from a permission, no?

File: apps/dashboard/app/(app)/@breadcrumb/authorization/roles/[roleId]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+export const runtime = "edge";
+
+type PageProps = {
+  params: { roleId: string };
+};
+
+async function AsyncPageBreadcrumb(props: PageProps) {
+  const tenantId = getTenantId();
+
+  const workspace = await db.query.workspaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/@breadcrumb/authorization/roles/[roleId]/page.tsx:25]: same here, it's a huge query for a simple lookup and not cached

File: apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/logs/page.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+export const runtime = "edge";
+
+type PageProps = {
+  params: { namespaceId: string };
+};
+
+async function AsyncPageBreadcrumb(props: PageProps) {
+  const tenantId = getTenantId();
+
+  const namespace = await db.query.ratelimitNamespaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/logs/page.tsx:25]: not cached

File: apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/overrides/page.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+export const runtime = "edge";
+
+type PageProps = {
+  params: { namespaceId: string };
+};
+
+async function AsyncPageBreadcrumb(props: PageProps) {
+  const tenantId = getTenantId();
+
+  const namespace = await db.query.ratelimitNamespaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/overrides/page.tsx:25]: not cached

File: apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export const runtime = "edge";
+
+type PageProps = {
+  params: { namespaceId: string };
+};
+
+async function AsyncPageBreadcrumb(props: PageProps) {
+  const tenantId = getTenantId();
+
+  const namespace = await db.query.ratelimitNamespaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/page.tsx:25]: not cached

File: apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/settings/page.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+export const runtime = "edge";
+
+type PageProps = {
+  params: { namespaceId: string };
+};
+
+async function AsyncPageBreadcrumb(props: PageProps) {
+  const tenantId = getTenantId();
+
+  const namespace = await db.query.ratelimitNamespaces.findFirst({
```
Comment (chronark) [inline apps/dashboard/app/(app)/@breadcrumb/ratelimits/[namespaceId]/settings/page.tsx:25]: not cached

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx (hunk 1 above)
Comment (guilhermerodz) [reply apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx:25]: that's true, I should remove that

**Comment by guilhermerodz** [comment]
@chronark have we already cached any other route for breadcrumbs? just to take as an example

File: apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx (hunk 1 above)
Comment (chronark) [reply apps/dashboard/app/(app)/@breadcrumb/authorization/permissions/[permissionId]/page.tsx:25]: yeah, do it like you did for the ratelimit namespaces, that query is good

**Comment by chronark** [comment]
[https://github.com/unkeyed/unkey/blob/2fc54aa9aabd66bbe17e26df47193ee94ed6d109/apps/dashboard/app/(app)/%40breadcrumb/apis/%5BapiId%5D/page.tsx#L26](https://github.com/unkeyed/unkey/blob/2fc54aa9aabd66bbe17e26df47193ee94ed6d109/apps/dashboard/app/(app)/%40breadcrumb/apis/%5BapiId%5D/page.tsx#L26)
yes all of the existing ones

**Pull request merged**